diff --git a/src/daemon/pipewire.conf.in b/src/daemon/pipewire.conf.in
index 697bf09..12d07a3 100644
--- a/src/daemon/pipewire.conf.in
+++ b/src/daemon/pipewire.conf.in
@@ -294,8 +294,8 @@ context.exec = [
     # but it is better to start it as a systemd service.
     # Run the session manager with -h for options.
     #
-    @sm_comment@{ path = "@session_manager_path@" args = "@session_manager_args@"
-    @sm_comment@  condition = [ { exec.session-manager = null } { exec.session-manager = true } ] }
+    { path = "@session_manager_path@" args = "@session_manager_args@"
+    condition = [ { exec.session-manager = null } { exec.session-manager = true } ] }
     #
     # You can optionally start the pulseaudio-server here as well
     # but it is better to start it as a systemd service.
