diff --git a/spa/plugins/alsa/alsa-udev.c b/spa/plugins/alsa/alsa-udev.c
index 3048d73..8792a5c 100644
--- a/spa/plugins/alsa/alsa-udev.c
+++ b/spa/plugins/alsa/alsa-udev.c
@@ -162,9 +162,6 @@ static unsigned int get_card_nr(struct impl *this, struct udev_device *udev_devi
 	if ((str = udev_device_get_property_value(udev_device, "SOUND_CLASS")) && spa_streq(str, "modem"))
 		return SPA_ID_INVALID;
 
-	if (udev_device_get_property_value(udev_device, "SOUND_INITIALIZED") == NULL)
-		return SPA_ID_INVALID;
-
 	if ((str = udev_device_get_property_value(udev_device, "DEVPATH")) == NULL)
 		return SPA_ID_INVALID;
 
@@ -891,7 +888,7 @@ static void impl_on_fd_events(struct spa_source *source)
 
 	start_inotify(this);
 
-	if (spa_streq(action, "change")) {
+	if (spa_streq(action, "change") || spa_streq(action, "add")) {
 		process_card(this, ACTION_ADD, udev_device);
 	} else if (spa_streq(action, "remove")) {
 		process_card(this, ACTION_REMOVE, udev_device);
@@ -958,7 +955,7 @@ static int enum_cards(struct impl *this)
 
 	for (udev_devices = udev_enumerate_get_list_entry(enumerate); udev_devices;
 			udev_devices = udev_list_entry_get_next(udev_devices)) {
-		struct udev_device *udev_device;
+		struct udev_device *udev_device, *pdev;
 
 		udev_device = udev_device_new_from_syspath(this->udev,
 		                                           udev_list_entry_get_name(udev_devices));
@@ -967,6 +964,10 @@ static int enum_cards(struct impl *this)
 
 		process_card(this, ACTION_ADD, udev_device);
 
+		pdev = udev_device_get_parent(udev_device);
+		if (pdev)
+			process_card(this, ACTION_ADD, pdev);
+
 		udev_device_unref(udev_device);
 	}
 	udev_enumerate_unref(enumerate);
