diff --git a/subprojects/wireplumber/src/config/bluetooth.lua.d/50-bluez-config.lua b/subprojects/wireplumber/src/config/bluetooth.lua.d/50-bluez-config.lua
index d5727d3..a983b3a 100644
--- a/subprojects/wireplumber/src/config/bluetooth.lua.d/50-bluez-config.lua
+++ b/subprojects/wireplumber/src/config/bluetooth.lua.d/50-bluez-config.lua
@@ -43,7 +43,7 @@ bluez_monitor.properties = {
   -- pipewire and wireplumber).
   -- This requires access to the D-Bus user session; disable if you are running
   -- a system-wide instance of wireplumber.
-  ["with-logind"] = true,
+  ["with-logind"] = false,
 }
 
 bluez_monitor.rules = {
diff --git a/subprojects/wireplumber/src/config/main.lua.d/50-alsa-config.lua b/subprojects/wireplumber/src/config/main.lua.d/50-alsa-config.lua
index 059191c..a345b2f 100644
--- a/subprojects/wireplumber/src/config/main.lua.d/50-alsa-config.lua
+++ b/subprojects/wireplumber/src/config/main.lua.d/50-alsa-config.lua
@@ -10,7 +10,7 @@ alsa_monitor.properties = {
   -- Reserve devices via org.freedesktop.ReserveDevice1 on D-Bus
   -- Disable if you are running a system-wide instance, which
   -- doesn't have access to the D-Bus user session
-  ["alsa.reserve"] = true,
+  ["alsa.reserve"] = false,
   --["alsa.reserve.priority"] = -20,
   --["alsa.reserve.application-name"] = "WirePlumber",
 
diff --git a/subprojects/wireplumber/src/config/main.lua.d/50-default-access-config.lua b/subprojects/wireplumber/src/config/main.lua.d/50-default-access-config.lua
index 4ad3d57..39d9a19 100644
--- a/subprojects/wireplumber/src/config/main.lua.d/50-default-access-config.lua
+++ b/subprojects/wireplumber/src/config/main.lua.d/50-default-access-config.lua
@@ -4,7 +4,7 @@ default_access.properties = {
   -- Enable the use of the flatpak portal integration.
   -- Disable if you are running a system-wide instance, which
   -- doesn't have access to the D-Bus user session
-  ["enable-flatpak-portal"] = true,
+  ["enable-flatpak-portal"] = false,
 }
 
 default_access.rules = {
