--- a/config.h.in
+++ b/config.h.in
@@ -191,6 +191,15 @@
 #ifndef _POSIX_SOURCE
 # undef _POSIX_SOURCE
 #endif
+/* taken from glibc unistd.h and fixes musl */
+#ifndef TEMP_FAILURE_RETRY
+#define TEMP_FAILURE_RETRY(expression) \
+  (__extension__                                                              \
+    ({ long int __result;                                                     \
+       do __result = (long int) (expression);                                 \
+       while (__result == -1L && errno == EINTR);                             \
+       __result; }))
+#endif
 /* Define to 2 if needed for POSIX-compatible behavior.  */
 #ifndef _POSIX_1_SOURCE
 # undef _POSIX_1_SOURCE
