diff --git a/src/koga/units.lisp b/src/koga/units.lisp
index 808cebd..5145888 100644
--- a/src/koga/units.lisp
+++ b/src/koga/units.lisp
@@ -44,6 +44,7 @@
     (append-ldlibs configuration (run-program-capture (list llvm-config "--system-libs")))
     (append-ldlibs configuration (run-program-capture (list llvm-config "--libs")))
     #+freebsd (append-ldlibs configuration "-lexecinfo")
+    #+linux   (append-ldlibs configuration "-lexecinfo")
     (when (ld configuration)
       (append-ldflags configuration (format nil "-fuse-ld=~(~a~)" (ld configuration))))
     (append-ldflags configuration "-pthread -fvisibility=default -rdynamic")))
