diff --git a/src/core/foundation.cc b/src/core/foundation.cc
index f32f629..3d41ed0 100644
--- a/src/core/foundation.cc
+++ b/src/core/foundation.cc
@@ -1669,7 +1669,7 @@ void maybe_register_symbol_using_dladdr_ep(void *functionPointer, size_t size, c
         return; // This means it's a virtual method.
       global_pointerCount++;
       Dl_info info;
-#if defined(_TARGET_OS_LINUX)
+#if defined(_TARGET_OS_LINUX) && defined(__GLIBC__)
       const Elf64_Sym *extra_info;
       int ret = dladdr1(functionPointer, &info, (void **)&extra_info, RTLD_DL_SYMENT);
 #else
