diff --git a/src/AST.cpp b/src/AST.cpp
index 777aff7..df0fc12 100644
--- a/src/AST.cpp
+++ b/src/AST.cpp
@@ -242,7 +242,7 @@ Decl* C2FFIASTConsumer::make_decl(const clang::VarDecl* d, bool is_toplevel)
     if(d->hasInit()) {
         if(!d->getType()->isDependentType()) {
             clang::EvaluatedStmt* stmt = d->ensureEvaluatedStmt();
-            clang::Expr*          e    = clang::cast<clang::Expr>(stmt->Value);
+            clang::Expr*          e    = clang::cast<clang::Expr*>(stmt->Value);
             if(!e->isValueDependent() && ((v = d->evaluateValue()) || (v = d->getEvaluatedValue()))) {
                 if(v->isLValue()) {
                     clang::APValue::LValueBase base = v->getLValueBase();
