diff --git a/Makefile b/Makefile
index e91beca..5edc783 100644
--- a/Makefile
+++ b/Makefile
@@ -91,7 +91,7 @@ share/man/.man-pages.stamp: $(HELP_ALL:=.md) ./man-template.html bin/md2roff
 share/man/man1/hub.1.md:
 	true
 
-install: bin/hub man-pages
+install: bin/hub
 	bash < script/install.sh
 
 clean:
diff --git a/script/install.sh b/script/install.sh
index c3230d9..845d23f 100644
--- a/script/install.sh
+++ b/script/install.sh
@@ -16,7 +16,7 @@ fi
 prefix="${PREFIX:-$prefix}"
 prefix="${prefix:-/usr/local}"
 
-for src in bin/hub share/man/*/*.1 share/doc/*/*.html share/vim/vimfiles/*/*.vim; do
+for src in bin/hub; do
   dest="${DESTDIR}${prefix}/${src}"
   mkdir -p "${dest%/*}"
   [[ $src == share/* ]] && mode="644" || mode=755
